src/env.cc

```diff
@@ -915,10 +928,6 @@ Environment::~Environment() {
       addon.Close();
     }
   }
-
-  for (auto realm : shadow_realms_) {
-    realm->OnEnvironmentDestruct();
```

Comment:
Can you add a test that creates a bunch of shadow realms and then generates a heap snapshot? I think that's how I ran into the shadow realms outliving shadow realms issue and is why this is necessary.

Reply:
Added a new test `test/pummel/test-heapdump-shadow-realm.js`.